cmd/contour/serve.go

```diff
@@ -361,6 +478,7 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 	// step 13. create grpc handler and register with workgroup.
 	g.Add(func(stop <-chan struct{}) error {
 		log := log.WithField("context", "grpc")
+		<-leaderOK
```

Comment:
As above, we need to select on leaderOK or stop because this function might need to stop before it is elected.